### PR TITLE
[syncd-rpc docker] Fix issue: ptf_nn_agent isn't able to start in syncd-rpc docker on buster

### PR DESCRIPTION
**Pull request by stephenxs**
**- Why I did it**
Fix issue: ptf_nn_agent isn't able to start in syncd-rpc docker on buster.

**- How I did it**
The issue is fixed by installing python-dev, cffi and nnpy for python 2 explicitly.

**- How to verify it**
Run copp test on RPC image.

**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [x] 202012

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by liat-grozovik — APPROVED** [review]

**Comment by liat-grozovik** [comment]
retest vsimage please

File: platform/mellanox/docker-syncd-mlnx-rpc/Dockerfile.j2 [hunk 1]
```diff
@@ -45,9 +46,10 @@ RUN apt-get update \
  && cd ..               \
  && rm -fr nanomsg-1.0.0 \
  && rm -f 1.0.0.tar.gz  \
- && pip install cffi==1.7.0    \
- && pip install --upgrade cffi==1.7.0 \
- && pip install nnpy    \
+ && pip2 install cffi==1.7.0    \
+ && pip2 install --upgrade cffi==1.7.0 \
+ && pip2 install wheel  \
+ && pip2 install nnpy    \
```
Comment (lguohan) [inline platform/mellanox/docker-syncd-mlnx-rpc/Dockerfile.j2:52]: is this change only related to mellanox syncd rpc?

I think it should be applicable to all syncd-rpc

Comment (stephenxs) [reply platform/mellanox/docker-syncd-mlnx-rpc/Dockerfile.j2:52]: Looks like this part is the same among all vendors. So it's likely to be applicable to all syncd-rpc dockers.
As they are defined on a per-vendor basis I think separate PRs should be opened for each vendor.

Comment (lguohan) [reply platform/mellanox/docker-syncd-mlnx-rpc/Dockerfile.j2:52]: can you fix all the vendors? later, we can unified the rpc docker for all vendors (this can be a separate pr).

Comment (stephenxs) [reply platform/mellanox/docker-syncd-mlnx-rpc/Dockerfile.j2:52]: I can modify the code for all the vendors but I’m not able to compile and test. Is it ok?

Comment (lguohan) [reply platform/mellanox/docker-syncd-mlnx-rpc/Dockerfile.j2:52]: yes

Comment (lguohan) [reply platform/mellanox/docker-syncd-mlnx-rpc/Dockerfile.j2:52]: there seems some docker is still stretch based, does it affect by this issue? why is this mellanox rpc docker is broken? what was the trigger?

Comment (stephenxs) [reply platform/mellanox/docker-syncd-mlnx-rpc/Dockerfile.j2:52]: 1. The Python has been migrated to version 3
2. cffi build failed because neither Python2-dev nor python3-dev is installed.
3. As ptf_nn_agent and sdk API are using Python2, we need to install those debs for Python 2.

Comment (stephenxs) [reply platform/mellanox/docker-syncd-mlnx-rpc/Dockerfile.j2:52]: Hi @lguohan , 
Could you please help check whether this issue also occurred on other vendor's RPC docker?
This can be done by checking
- whether there is an error when installing nnpy in docker-<vendor>-mlnx-rpc.gz.log, or
- whether the ptf_nn_agent is running.
If it does, I suggest having this bug fixed by this PR.
Thanks

Comment (lguohan) [reply platform/mellanox/docker-syncd-mlnx-rpc/Dockerfile.j2:52]: i believe this is because of python3 migration. can you check the syncd docker of the platform is buster-based or not? if it is buster based, then we need your patch. if it is stretch based, then we do not need this patch.

File: platform/broadcom/docker-syncd-brcm-rpc/Dockerfile.j2 [hunk 2]
```diff
@@ -21,6 +21,7 @@ RUN apt-get update \
  && apt-get -y install  \
     net-tools           \
     python-pip          \
+    python-setuptools   \
```
Comment (lguohan) [inline platform/broadcom/docker-syncd-brcm-rpc/Dockerfile.j2:24]: brcm should be buster based.

File: platform/mellanox/docker-syncd-mlnx-rpc/Dockerfile.j2 (hunk 1 above)
Comment (stephenxs) [reply platform/mellanox/docker-syncd-mlnx-rpc/Dockerfile.j2:52]: You’re correct. Only Broadcom and we use buster based syncs rpc docker.
code updated accordingly.

**Comment by stephenxs** [comment]
retest this please

**Comment by lguohan** [comment]
docker-syncd-bfn
docker-syncd-centec

are also buster baed.

**Comment by stephenxs** [comment]
> docker-syncd-bfn
> docker-syncd-centec
> 
> are also buster baed.

Indeed. These platforms were updated just after I had forked the branch of this PR.
I rebased and add them. Now, in the following platforms, syncd rpc image are buster based:
- Mellanox
- Broadcom
- Barefoot
- Centec and Centec arm64

**Review by lguohan — APPROVED** [review]

**Review by liat-grozovik — APPROVED** [review]
